### PR TITLE
Pool search clusters

### PR DESCRIPTION
**Pull request by jcorbin**
r @Raynos @kriskowal

### CODE REVIEW

File: node/test/lib/alloc-cluster.js [hunk 1]
```diff
@@ -156,3 +157,34 @@ allocCluster.test = function testCluster(desc, opts, t) {
...
+function ClusterPool(setup) {
+    var self = this;
+    self.free = [];
+    self.setup = setup;
+}
+
+ClusterPool.prototype.get = function getCluster(callback) {
+    var self = this;
+    if (self.free.length) {
+        callback(null, self.free.shift());
```
Comment (kriskowal) [inline node/test/lib/alloc-cluster.js]: zalgo has left the building

Comment (jcorbin) [inline node/test/lib/alloc-cluster.js]: #itsjustatest #nothingtoseehere #movealongnow

File: node/test/streaming_bisect.js [hunk 2]
```diff
@@ -162,7 +166,7 @@ test('bisection test', function t(assert) {
         };
     }), function done(err) {
         if (err && err !== firstStop) assert.ifError(err, 'no final error');
-        assert.end();
+        search.clusterPool.destroy(assert.end);
```
Comment (kriskowal) [inline node/test/streaming_bisect.js]: winful

File: node/test/streaming_bisect.js [hunk 3]
```diff
@@ -214,26 +238,31 @@ TestStreamSearch.prototype.init = function init() {
...
+    var cluster = null;
+
+    self.clusterPool.get(gotCluster);
+
+    function gotCluster(err, clus) {
+        if (err) {
+            assert.end(err);
+            return;
+        }
+        cluster = clus;
```
Comment (kriskowal) [inline node/test/streaming_bisect.js]: Unclear why this variable is rebound

Comment (jcorbin) [inline node/test/streaming_bisect.js]: Is needed in a sibling function later on... #onlysomuchinthispr

File: node/test/streaming_bisect.js [hunk 4]
```diff
@@ -254,9 +283,14 @@ TestStreamSearch.prototype.test = function test(state, assert) {
                     }]
                 }]
             });
-            cluster.destroy(assert.end);
+            if (!assert._ok) {
+                cluster.destroy(assert.end);
```
Comment (jcorbin) [inline node/test/streaming_bisect.js]: see here now

**Comment by kriskowal** [comment]
:ok:

**Pull request merged**